lib/src/home/widgets/code_field.dart

```diff
@@ -230,4 +136,110 @@ class CodeFieldWidgetState extends State<CodeFieldWidget> {
       ),
     );
   }
+
+  @override
+  void didUpdateWidget(CodeFieldWidget oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    widget.controller.removeListener(_updateLineNumber);
+    widget.controller.addListener(_updateLineNumber);
+  }
+
+  @override
+  void dispose() {
+    _numberController.dispose();
+    _fieldController.dispose();
+    _keyboardVisibilitySubscription?.cancel();
+
+    widget.controller.removeListener(_updateLineNumber);
+
+    super.dispose();
+  }
+
+  @override
+  void initState() {
+    super.initState();
+
+    _scrollControllers = LinkedScrollControllerGroup();
+    _numberController = _scrollControllers.addAndGet();
+    _fieldController = _scrollControllers.addAndGet();
+
+    widget.controller.addListener(_updateLineNumber);
+
+    _listViewWidth = widget.initialListViewWidth;
+
+    _focusNode = widget.focusNode ?? FocusNode();
+    _focusNode!.attach(context, onKey: _onKey);
+
+    _lineCount = '\n'.allMatches(widget.controller.text).length + 1;
+  }
+
+  int _getDigits(int num) {
+    var count = 0;
+    while (num != 0) {
+      num = (num / 10).floor();
+      count++;
+    }
+    return count;
+  }
+
+  Color? _getLineNumberColor(
+      bool selected, Map<String, TextStyle>? theme, String? rootKey) {
+    if (theme != null && rootKey != null) {
+      return selected
+          ? theme[rootKey]!.color
+          : theme[rootKey]!.color?.withAlpha(128);
```

Comment:
Those look the most sensible to me.
I didn't check if theme can ever be null here - I think it shouldn't, but the previous version is still included as a fallback.